### PR TITLE
[13.0] helpdesk_mgmt: add text input in ticket list searchbar

### PR DESCRIPTION
**Pull request by luffah**
See https://github.com/OCA/helpdesk/issues/64 as original issue.
See https://github.com/OCA/helpdesk/pull/69 for v11.0.
See https://github.com/OCA/helpdesk/pull/141 for v12.0.

This PR add search bar text input to portal ticket list.
Searchable fields are : name; description; user; category; (optionally message content)
2 meta fields extend searchable fields : All (lookup all searchable fields); Content (name + description + optionally message content) 

Searching in message content is enabled by installing the module [base_search_mail_content](https://github.com/OCA/social/tree/13.0/base_search_mail_content).

### CODE REVIEW

File: helpdesk_mgmt/controllers/myaccount.py [hunk 1]
```diff
@@ -27,14 +29,21 @@ def _helpdesk_ticket_check_access(self, ticket_id):
             raise
         return ticket_sudo
 
+    # FIXME module to migrate dans oca/social
+    def _message_content_field_exists(self):
+        base_search_module = request.env['ir.module.module'].sudo().search([
+            ('name', '=', 'base_search_mail_content')])
```
Comment (flachica) [inline helpdesk_mgmt/controllers/myaccount.py]: You must put this dependence in <repo_root>/oca_dependencies.txt

**Review by flachica — CHANGES_REQUESTED** [review]

**Comment by flachica** [comment]
If yoy run pre-commit run -a you can solve [this](https://travis-ci.com/github/OCA/helpdesk/jobs/360492181#L278)

Functional test OK

Thanks for your work

**Comment by luffah** [comment]
> If yoy run pre-commit run -a you can solve [this](https://travis-ci.com/github/OCA/helpdesk/jobs/360492181#L278)

yoy ? is it a tool ? ;)

**Comment by emagdalenaC2i** [comment]
@luffah See https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0

**Comment by luffah** [comment]
I was looking again on this PR.
If i understand well, it is blocked because a test is missing.

File: helpdesk_mgmt/controllers/myaccount.py (hunk 1 above)
Comment (luffah) [reply helpdesk_mgmt/controllers/myaccount.py]: added base_search_mail_content in oca_dependencies.txt

**Comment by luffah** [comment]
merged with most recent changes…
oh crap, i shall redo a new PR…

**Pull request closed without merging**